cumulusci/tasks/metadeploy.py

```diff
@@ -0,0 +1,131 @@
+import requests
+
+from cumulusci.core.config import TaskConfig
+from cumulusci.core.tasks import BaseTask
+from cumulusci.core.flowrunner import FlowCoordinator
+
+
+class BaseMetaDeployTask(BaseTask):
+    """Base class for tasks that talk to MetaDeploy's API."""
+
+    def _init_task(self):
+        metadeploy_service = self.project_config.keychain.get_service("metadeploy")
+        self.base_url = metadeploy_service.url
+        self.api = requests.Session()
+        self.api.headers["Authorization"] = "token {}".format(metadeploy_service.token)
+
+    def _call_api(self, method, path, collect_pages=False, **kwargs):
+        next_url = self.base_url + path
+        results = []
+        while next_url is not None:
+            response = self.api.request(method, next_url, **kwargs)
+            response.raise_for_status()
+            response = response.json()
+            if "links" in response and collect_pages:
+                results.extend(response["data"])
+                next_url = response["links"]["next"]
+            else:
+                return response
+        return results
+
+
+class Publish(BaseMetaDeployTask):
+    """Publishes an installation plan to MetaDeploy.
+    """
+
+    task_options = {
+        "flow": {"description": "Name of flow to publish", "required": True},
+        "product_id": {
```

Comment:
Could we use the github repo url on the Product model to look this up by default and only require this if there's more than one Product for the repo?

Reply:
Yeah, we just don't have a way to filter the API yet so I figured hardcoding this was simpler.